cse-lib/data/src/main/java/com/farao_community/farao/cse/data/ntc/DailyNtcDocument.java

```diff
@@ -38,8 +39,12 @@ Map<String, LineInformation> getLineInformationPerLineId(Predicate<TLine> lineSe
                     .collect(Collectors.toMap(
                         TLine::getCode,
                         tLine -> {
-                            TNTC tNtc = NtcUtil.getTNtcFromLine(targetDateTime, tLine);
-                            return new LineInformation(tLine.getCNtc().value(), tNtc.getType(), tNtc.getV().doubleValue());
+                            Optional<TNTC> optionalTntc = NtcUtil.getTNtcFromLineFromNtcRedFile(targetDateTime, tLine);
+                            if (optionalTntc.isPresent()) {
+                                return Optional.of(new LineInformation(tLine.getCNtc().value(), optionalTntc.get().getType(), optionalTntc.get().getV().doubleValue()));
+                            } else {
+                                return Optional.empty();
```

Comment:
Is it necessary to put all line-codes in the Map, even if they are not associated with any data?
Putting in it only the elements that are associated with a real value would make the code lighter, as you wouldn't have to deal with Optionals everywhere.